addons/website/models/ir_ui_view.py

```diff
@@ -132,27 +132,24 @@ def get_related_views(self, key, bundles=False):
         self = self.with_context(website_id=self.env['website'].get_current_website().id)
         return super(View, self).get_related_views(key, bundles=bundles)
 
-    @api.multi
-    def _sort_suitability_key(self):
-        """ Key function to sort views by descending suitability
-            Suitability of a view is defined as follow:
-                * if the view and request website_id are matched
-                * then if the view has no set website
+    def filter_duplicate(self):
+        """ Filter current recordset only keeping the most suitable view per distinct key.
+            Every non-accessible view will be removed from the set:
+              * In non website context, every view with a website will be removed
+              * In a website context, every view from another website
         """
-        self.ensure_one()
-        context_website_id = self.env.context.get('website_id', None)
-        website_id = self.website_id.id or 0
-        different_website = context_website_id != website_id
-        return (different_website, website_id)
+        current_website_id = self._context.get('website_id')
```

Comment:
why not use self.env.context instead of this private getter????

Reply:
I did not change that line just move the code but I can change it?